File: backend/app/schemas/ai.py
from typing import List, Optional, Dict, Any, Union
from uuid import UUID
from pydantic import BaseModel, Field


class CampaignGenerationRequest(BaseModel):
    campaign_id: Optional[UUID] = None
    product_name: Optional[str] = "Product Launch"
    product_description: str = Field(..., description="Detailed description of product or service")
    industry: Optional[str] = "Technology & SaaS"
    target_audience: Optional[str] = "B2B Decision Makers"
    budget: Optional[str] = "$5,000 - $15,000 / month"
    goal: Optional[str] = "Lead Generation & Registrations"
    tone: Optional[str] = "Professional & Authoritative"
    target_channels: Optional[List[str]] = Field(default_factory=lambda: ["Google", "Facebook", "Instagram", "LinkedIn"])


class BudgetBreakdown(BaseModel):
    google: str = Field(default="35%", description="Google Ads budget share")
    facebook: str = Field(default="25%", description="Facebook Ads budget share")
    instagram: str = Field(default="20%", description="Instagram Ads budget share")
    linkedin: str = Field(default="20%", description="LinkedIn Ads budget share")


class CampaignGenerationResponse(BaseModel):
    buyer_persona: Union[str, Dict[str, Any]] = Field(..., description="Buyer persona description or breakdown")
    persona: Optional[Union[str, Dict[str, Any]]] = Field(default=None, description="Alias for buyer persona")
    marketing_strategy: Union[str, Dict[str, Any]] = Field(..., description="Core marketing strategy and messaging")
    google_ads: List[Any] = Field(default_factory=list, description="Google Ads list")
    facebook_ads: List[Any] = Field(default_factory=list, description="Facebook Ads list")
    instagram_ads: List[Any] = Field(default_factory=list, description="Instagram Ads list")
    linkedin_ads: List[Any] = Field(default_factory=list, description="LinkedIn Ads list")
    keywords: List[str] = Field(default_factory=list, description="SEO & PPC Keywords")
    hashtags: List[str] = Field(default_factory=list, description="Social Media Hashtags")
    budget_breakdown: Union[BudgetBreakdown, Dict[str, Any]] = Field(..., description="Budget breakdown across channels")
    status: str = Field(default="completed", description="Generation status")


class ContentRefinementRequest(BaseModel):
    channel: str
    current_text: str
    instructions: str

File: frontend/src/types/ai.ts
export interface BudgetBreakdown {
  google: string
  facebook: string
  instagram: string
  linkedin: string
}

export interface GoogleAd {
  headline_1?: string
  headline_2?: string
  headline_3?: string
  headline?: string
  description_1?: string
  description_2?: string
  description?: string
  call_to_action?: string
}

export interface FacebookAd {
  primary_text: string
  headline: string
  description?: string
  call_to_action?: string
}

export interface InstagramAd {
  caption: string
  visual_concept?: string
  call_to_action?: string
}

export interface LinkedInAd {
  post_text?: string
  headline?: string
  target_job_titles?: string[]
  call_to_action?: string
}

export interface CampaignGenerationRequest {
  campaign_id?: string
  product_name?: string
  product_description: string
  industry?: string
  target_audience?: string
  budget?: string
  goal?: string
  tone?: string
  target_channels?: string[]
}

export interface CampaignGenerationResponse {
  buyer_persona: any
  persona?: any
  marketing_strategy: any
  google_ads: GoogleAd[]
  facebook_ads: FacebookAd[]
  instagram_ads: InstagramAd[]
  linkedin_ads: LinkedInAd[]
  keywords: string[]
  hashtags: string[]
  budget_breakdown: BudgetBreakdown | Record<string, string>
  status: string
}

export interface ContentRefinementRequest {
  channel: string
  current_text: string
  instructions: string
}

export interface ContentRefinementResponse {
  refined_text: string
}
